pkg/devfile/parser/data/v2/components_test.go

```diff
@@ -403,3 +404,114 @@ func TestGetDevfileVolumeComponents(t *testing.T) {
 	}
 
 }
+
+func TestDeleteComponents(t *testing.T) {
+
+	tests := []struct {
+		name              string
+		componentToDelete string
+		components        []v1.Component
+		wantComponents    []v1.Component
+		wantErr           bool
+	}{
+		{
+			name:              "Volume Component with mounts",
```

Comment:
since we do pure deletion without considering the reference, we should rename the test case to avoid confusion